Yasgui/packages/yasqe/src/index.ts

```diff
@@ -492,6 +499,81 @@ export class Yasqe extends CodeMirror {
       (this.keyboardShortcutsButton as any).items = [...this.config.keyboardShortcutDescriptions];
     }
   }
+
+  private drawAbortQueryButton() {
+    if (!(this.config.onAbortQuery instanceof Function)) {
+      return;
+    }
+
+    const yasqeFooterButtons = document.createElement("div");
+    addClass(yasqeFooterButtons, "yasqe-footer-buttons");
+
+    this.abortQueryButton = document.createElement("div");
+    addClass(this.abortQueryButton, "abort-button");
+
+    this.abortQueryButton.onclick = () => {
+      if (this.isQueryAborted) {
+        return;
+      }
+      this.isQueryAborted = true;
+      const req = this.req;
+      this.abortQuery();
+      if (this.config.onAbortQuery instanceof Function) {
```

Comment:
I think that this event onAbortQuery should renamed to `onQueryAborted` because we actually call it after the abort query operation is already triggered.